src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+
+using Microsoft.PowerShell.Commands;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal.Json
+{
+    /// <summary>
+    /// Class that defines Dsc cache entries.
+    /// </summary>
+    internal class DscClassCacheEntry
+    {
+        /// <summary>
+        /// Store the RunAs Credentials that this DSC resource will use.
+        /// </summary>
+        public DSCResourceRunAsCredential DscResRunAsCred { get; set; }
+
+        /// <summary>
+        /// If we have implicitly imported this resource, we will set this field to true. This will
+        /// only happen to InBox resources.
+        /// </summary>
+        public bool IsImportedImplicitly { get; set; }
+
+        /// <summary>
+        /// A CimClass instance for this resource.
+        /// </summary>
+        public PSObject CimClassInstance { get; set; }
+
+        /// <summary>
+        /// Initializes variables with default values.
+        /// </summary>
+        public DscClassCacheEntry() : this(DSCResourceRunAsCredential.Default, false, null) { }
+
+        /// <summary>
+        /// Initializes all values.
+        /// </summary>
+        /// <param name="aDSCResourceRunAsCredential"></param>
+        /// <param name="aIsImportedImplicitly"></param>
+        /// <param name="aCimClassInstance"></param>
+        public DscClassCacheEntry(DSCResourceRunAsCredential aDSCResourceRunAsCredential, bool aIsImportedImplicitly, PSObject aCimClassInstance)
+        {
+            DscResRunAsCred = aDSCResourceRunAsCredential;
+            IsImportedImplicitly = aIsImportedImplicitly;
+            CimClassInstance = aCimClassInstance;
+        }
```

Comment:
These parameters are strangely named. Consider:

```suggestion
        /// <summary>
        /// Initializes all values.
        /// </summary>
        /// <param name="dscResourceRunAsCredential"></param>
        /// <param name="isImportedImplicitly"></param>
        /// <param name="cimClassInstance"></param>
        public DscClassCacheEntry(DSCResourceRunAsCredential dscResourceRunAsCredential, bool isImportedImplicitly, PSObject cimClassInstance)
        {
            DscResRunAsCred = dscResourceRunAsCredential;
            IsImportedImplicitly = isImportedImplicitly;
            CimClassInstance = cimClassInstance;
        }
```

Reply:
updated.